* Only "pop" subtree params if they had previously been "pushed", and 
  afterwards reset the "pushed" value to 0 again. Similarly only set the
  PLAY_TREE_RND_PLAYED flag if the entry had been pushed before.
diff -r 33007f27d435 playtree.c
--- a/playtree.c	Sun Apr 30 16:54:30 2006 +0200
+++ b/playtree.c	Sun Apr 30 16:54:31 2006 +0200
@@ -710,8 +710,9 @@ play_tree_iter_up_step(play_tree_iter_t*
   iter->tree = iter->tree->parent;
 
   // Pop subtree params
-  if(iter->config) {
+  if(iter->config && iter->entry_pushed > 0) {
     m_config_pop(iter->config);
+    iter->entry_pushed = 0;
     if(iter->mode == PLAY_TREE_ITER_RND)
       iter->tree->flags |= PLAY_TREE_RND_PLAYED;
   }
